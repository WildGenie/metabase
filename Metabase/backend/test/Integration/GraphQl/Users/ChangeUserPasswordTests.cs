using FluentAssertions;
using System.IO;
using System.Threading.Tasks;
using Snapshooter.Xunit;
using Xunit;
using System.Collections.Generic;

namespace Metabase.Tests.Integration.GraphQl.Users
{
    public sealed class ChangeUserPasswordTests
      : UserIntegrationTests
    {
        [Fact]
        public async Task ValidDataOfRegisteredAndConfirmedAndLoggedInUser_ChangesUserPassword()
        {
            // Arrange
            var email = DefaultEmail;
            var password = DefaultPassword;
            await RegisterAndConfirmAndAuthorizeUser(
                email: email,
                password: password
            ).ConfigureAwait(false);
            // Act
            var response = await SuccessfullyQueryGraphQlContentAsString(
                File.ReadAllText("Integration/GraphQl/Users/ChangeUserPasswordTests/ValidDataOfRegisteredAndConfirmedAndLoggedInUser_ChangesUserPassword.graphql"),
                variables: new Dictionary<string, object?>
                {
                    ["currentPassword"] = password,
                    ["newPassword"] = "new" + password,
                    ["newPasswordConfirmation"] = "new" + password
                }
                ).ConfigureAwait(false);
            // Assert
            Snapshot.Match(
                response,
                matchOptions => matchOptions.Assert(fieldOptions =>
                 fieldOptions.Field<string>("data.changeUserPassword.user.id").Should().NotBeNullOrWhiteSpace()
                 )
                );
        }

        // TODO Other cases!
    }
}
