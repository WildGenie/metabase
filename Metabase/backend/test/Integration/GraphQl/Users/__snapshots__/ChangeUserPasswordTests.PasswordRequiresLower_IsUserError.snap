{"data":{"changeUserPassword":{"user":{"id":"VXNlcgpnMDFkMDIyN2NkZGVjNDE0OGFmNGNlYWQxMWE3MTZmNjM=","email":"john.doe@ise.fraunhofer.de"},"errors":[{"code":"PASSWORD_REQUIRES_LOWER","message":"Passwords must have at least one lowercase (\u0027a\u0027-\u0027z\u0027).","path":["input","password"]}]}}}
